src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -1206,10 +1207,31 @@ public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, Action<
     /// <inheritdoc cref="UseTelemetry(IGraphQLBuilder, Action{GraphQLTelemetryOptions}?)"/>
     public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, Action<GraphQLTelemetryOptions, IServiceProvider>? configure)
     {
+        OpenTelemetry.AutoInstrumentation.Initializer.Enabled = false;
         builder.Services.Configure(configure);
         builder.Services.TryRegister<IConfigureExecution, GraphQLTelemetryProvider>(ServiceLifetime.Singleton, RegistrationCompareMode.ServiceTypeAndImplementationType);
         return builder;
     }
+
+    /// <inheritdoc cref="UseTelemetry(IGraphQLBuilder, Action{GraphQLTelemetryOptions}?)"/>
+    public static IGraphQLBuilder UseTelemetry<TProvider>(this IGraphQLBuilder builder)
```

Comment:
How do you pass the `GraphQLTelemetryOptions`? And if `TProvider : GraphQLTelemetryProvider`, it may be useful to make `TOptions : GraphQLTelemetryOptions`, so the options can be extended to support additional configuration for the extended provider.

Reply:
Same issue as `ErrorInfoProvider` and other builder methods.  The user will have to configure options with `Configure()` to configure the options class they are using.

Reply:
Why no provide an overload with options?

Reply:
This scenario it might make sense to add `TOptions : GraphQLTelemetryOptions` since if someone wanted a fully custom telemetry provider they should use `ConfigureExecution<TProvider>`

Reply:
Whereas `IErrorInfoProvider` does not require any options; just the default implementation allows for certain options.  Users should be able to write their own implementation without necessarily using options.

Reply:
Or here we could just add two more overloads........

Reply:
I don't mind more overloads.  But realize that now someone needs to list both the provider class and options class:

```cs
b.UseTelemetry<MyProvider, MyProviderOptions>(o => { /* options */ });

// or
b.UseTelemetry<MyProvider>();
b.Configure(new MyProviderOptions { /* options */ });
```

Reply:
@gao-artur comments?

Reply:
It depends on how many overloads are you ready to provide 😁
There can be an overload with `TProvider` and `GraphQLTelemetryOptions` and another with `TProvider` and `TOptions`

Reply:
Sometimes simpler is better.  How about an xml comment that says how to configure options!?!?  Do we really need to add 4 more options here?

```cs
    public static IGraphQLBuilder UseTelemetry<TProvider>(this IGraphQLBuilder builder, Action<GraphQLTelemetryOptions> options)
    public static IGraphQLBuilder UseTelemetry<TProvider>(this IGraphQLBuilder builder, Action<GraphQLTelemetryOptions, IServiceProvider> options)
    public static IGraphQLBuilder UseTelemetry<TProvider, TOptions>(this IGraphQLBuilder builder, Action<TOptions> options)
    public static IGraphQLBuilder UseTelemetry<TProvider, TOptions>(this IGraphQLBuilder builder, Action<TOptions, IServiceProvider> options)
```

It seems like overkill.  Nobody has ever asked for more overloads for AddErrorInfoProvider

Reply:
It's not like it really hurts.  It's just a larger codebase to maintain.  Right now I think there are tests covering every overload (I didn't do the added ones in this PR yet).  Adding 4 more overloads means 4 more tests, too.

Reply:
I think these 4 are the required minimum. There may be a need for a factory method to create `TProvider`, but we can wait till someone requests it.

Reply:
We don't have similar methods for `UseMemoryCache` or `UseAutomaticPersistedQueries` either.

Reply:
Thinking again, who will need an `IServiceProvider` to construct options? I don't think I ever needed something like that.

Reply:
It's really useful when you want to inspect IHttpContext before setting options.  So for UseErrorInfoProvider you can use it to look for a specific header or IP address or similar before enabling detailed error reports.

Reply:
Ok, then. Let's keep only the first two?

Reply:
It's also useful when you set options from `IConfiguration` (environment variables, app settings or user secrets); you just pull the options class from DI.  But of course usually `IConfiguration` can be had prior and so it's not necessary to pull from DI.

Reply:
Well, basically I'd just do all 4 together on request.  But if you're requesting them then I'd just add all 4 now.

Reply:
Yeah, you are right. I recall now that I used this pattern when need to create options with dependencies
```c#
services.AddOptions<IConfiguration>()
    .Configure<GraphQLTelemetryOptions>((configuration, options) => { });
```

Reply:
Another way to think of it is that people can just `AddProvider<T>` and they should be quite familiar with DI and the fact that if you have an options class, they need to `AddSingleton` to register it

Reply:
Having these extra overloads should not be confusing for any user.

Reply:
> Well, basically I'd just do all 4 together on request. But if you're requesting them then I'd just add all 4 now.

Please add them. Because I just copied the `UseTelemetry` method and updated it to register the derived type, because I didn't find a simpler way to do that.